modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -1005,3 +1004,300 @@ function commerce_pos_report_build_journal_role_table(array $filters = array())
     );
   }
 }
+
+/**
+ * Builds a KPI Report result table.
+ *
+ * @param array $filters
+ *   An array of filters for the query. Can include any of the following:
+ *   'cashier' - The cashier who performed the transactions.
+ *   'payment_type' - The type of payment the transactions had.
+ *   'results_per_page' - The number of results per page.
+ *
+ * @return array
+ *   A render array for the result table.
+ */
+function commerce_pos_report_build_kpi_report_table(array $filters = array()) {
+  $header = array(
+      array('data' => t('Cashier Id'), 'field' => 't.cashier'),
+      array('data' => t('Cashier'), 'field' => 'cashier.name'),
+      array('data' => t('Net Sales')),
+      array('data' => t('Transactions')),
+      array('data' => t('Items Sold')),
+      array('data' => t('Unit per Trans (UPT)')),
+      array('data' => t('$ Per Transaction (DPT)')),
+  );
+
+  $query = db_select('commerce_pos_transaction', 't')
+    ->extend('PagerDefault')
+    ->extend('TableSort');
+
+  $query->fields('t', array('order_id', 'completed', 'uid', 'type'));
+  $query->addField('t', 'cashier', 'cashier_id');
+
+  // Cashier data
+  // We left join because legacy transactions may not have a cashier.
+  $query->leftJoin('commerce_pos_cashier', 'cashier', 'cashier.cashier_id = t.cashier');
+  $query->addField('cashier', 'name', 'cashier');
+
+  // Payment transaction data.
+  $query->join('commerce_payment_transaction', 'pt', 'pt.order_id = t.order_id');
+  $query->fields('pt', array('transaction_id'));
+
+  // Join order data.
+  $query->join('commerce_order', 'o', 'o.order_id = t.order_id');
+  $query->leftJoin('users', 'customer', 'customer.uid = o.uid');
+  $query->addField('customer', 'mail', 'customer_mail');
+
+  // Join order total data.
+  $query->join('field_data_commerce_order_total', 'ot', 't.order_id = ot.entity_id AND ot.entity_type = :commerce_order', array(
+    ':commerce_order' => 'commerce_order',
+  ));
+  $query->fields('ot', array(
+    'commerce_order_total_amount',
+    'commerce_order_total_currency_code',
+    'commerce_order_total_data',
+  ));
+
+  // Join line item data.
+  $query->join('commerce_line_item', 'l', 'l.order_id = t.order_id');
+  $query->addField('l', 'quantity');
+  $query->addExpression('COUNT(DISTINCT l.line_item_id)', 'item_count');
+
+  $line_item_types = commerce_product_line_item_types();
+  $query->condition('l.type', $line_item_types, 'IN');
+
+  // Make sure line item count is aggregated.
+  $query->groupBy('l.order_id');
+
+  /*
+   * As mysql do not support limits in sub queries
+   * create sub query for pagination
+   * and to limit records use a separate query.
+   */
+  $subQuery = db_select('commerce_pos_cashier', 'cpc')
```

Comment:
lowerCamelCase variables shouldn't be mixed with lower_underscore variables in a file. I prefer lower_underscore in module files. https://www.drupal.org/docs/develop/standards/coding-standards#naming